genie-web/src/main/java/com/netflix/genie/web/data/services/JobPersistenceService.java

```diff
@@ -273,11 +273,11 @@ void updateJobStatus(
      *
      * @param id The id of the job
      * @return true if its a v4 job
-     * @throws GenieJobNotFoundException If no job with {@code id} exists
+     * @throws GenieNotFoundException If no job with the given {@code id} exists
```

Comment:
Technically this was a new API and the other APIs are supporting old v3 job execution paths that were going to be deleted when migration was complete. I know you're adamant about your love for checked exceptions so I'm not going to argue that again cause it's not worth it but we should at least make this a checked "JobNotFoundException" so that we can be clear about what isn't found for APIs at higher levels that compound multiple actions. `GenieNotFoundException` as it bubbles up will be lost as to whether it was a job, cluster, command, file, etc.